src/MySqlConnector/Protocol/Serialization/ByteBufferWriter.cs

```diff
@@ -259,6 +259,27 @@ public static void WriteLengthEncodedString(this ByteBufferWriter writer, ReadOn
 		}
 #endif
 
+		public static void WriteLengthEncodedString(this ByteBufferWriter writer, StringBuilder builder)
+		{
+#if !NET45 && !NET461 && !NET471 && !NETSTANDARD1_3 && !NETSTANDARD2_0 && !NETSTANDARD2_1 && !NETCOREAPP2_1
+			var totalLength = 0;
+
+			foreach (var chunk in builder.GetChunks())
+			{
+				totalLength += Encoding.UTF8.GetByteCount(chunk.Span);
+			}
+
+			writer.WriteLengthEncodedInteger((ulong) totalLength);
+
+			foreach (var chunk in builder.GetChunks())
+			{
+				writer.Write(chunk.Span);
```

Comment:
There's kind of a latent bug here, in that this method assumes a complete UTF-16 `Span<char>`. Again, this will need to use an `Encoder` that maintains its own internal state across chunk boundaries.

Reply:
this was fixed in the further commits, thus I'm marking this as a resolved